src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserRoleService.cs

```diff
@@ -0,0 +1,307 @@
+                .CheckCanManageAllUsers()
+                .OnSuccess(_ => _usersAndRolesPersistenceHelper.CheckEntityExists<ApplicationUser, string>(userId))
+                .OnSuccess(async () =>
+                {
+                    var roleIds = await _usersAndRolesDbContext.UserRoles
+                        .Where(r => r.UserId == userId)
+                        .Select(r => r.RoleId)
+                        .ToListAsync();
+
+                    return await _usersAndRolesDbContext.Roles
+                        .Where(r => roleIds.Contains(r.Id))
+                        .OrderBy(r => r.Name)
+                        .Select(r => new RoleViewModel
+                        {
+                            Id = r.Id,
+                            Name = r.Name,
+                            NormalizedName = r.NormalizedName
+                        })
+                        .ToListAsync();
+                });
+        }
+
+        public async Task<Either<ActionResult, List<UserPublicationRoleViewModel>>> GetPublicationRoles(Guid userId)
+        {
+            return await _userService
+                .CheckCanManageAllUsers()
+                .OnSuccess(_ => _contentPersistenceHelper.CheckEntityExists<User>(userId))
+                .OnSuccess(async () =>
+                {
+                    return await _contentDbContext.UserPublicationRoles
+                        .Include(userPublicationRole => userPublicationRole.Publication)
+                        .Where(userPublicationRole => userPublicationRole.UserId == userId)
+                        .OrderBy(userPublicationRole => userPublicationRole.Publication.Title)
+                        .Select(userPublicationRole => new UserPublicationRoleViewModel
+                        {
+                            Id = userPublicationRole.Id,
+                            Publication = userPublicationRole.Publication.Title,
+                            Role = userPublicationRole.Role
+                        })
+                        .ToListAsync();
+                });
+        }
+
+        public async Task<Either<ActionResult, List<UserReleaseRoleViewModel>>> GetReleaseRoles(Guid userId)
+        {
+            return await _userService
+                .CheckCanManageAllUsers()
+                .OnSuccess(_ => _contentPersistenceHelper.CheckEntityExists<User>(userId))
+                .OnSuccess(async () =>
+                {
+                    var allReleaseRoles = await _contentDbContext.UserReleaseRoles
+                        .Include(userReleaseRole => userReleaseRole.Release)
+                        .ThenInclude(release => release.Publication)
+                        .Where(userReleaseRole => userReleaseRole.UserId == userId)
+                        .ToListAsync();
+
+                    var latestReleaseRoles = allReleaseRoles
+                        .Where(userReleaseRole => userReleaseRole.Release.Publication.IsLatestVersionOfRelease(
```

Comment:
Could you rewrite instead `.Where(userReleaseRole => !userReleaseRole.Release.Amendment)`?

Reply:
I'm not sure this is right because we do want to include amendments. If an amendment gets created and a user is assigned contributor on that amendment, it should show up in the table. Any roles they've got on previous versions of that Release should not be shown in the table.

Reply:
Gotcha!